api/schema.go

```diff
@@ -0,0 +1,50 @@
+package api
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+)
+
+type handlerSchema map[string]map[string]map[string]interface{}
+
+func getResourceSchema(name string) handlerSchema {
+	file, _ := os.Open("./api/schema/" + name + ".json")
+	defer file.Close()
+	var schemaObj handlerSchema
+	dec := json.NewDecoder(file)
+	if err := dec.Decode(&schemaObj); err != nil {
+		log.Fatal(err)
+	}
+
+	jtdToJsonSchema(schemaObj, "request", "params")
+	jtdToJsonSchema(schemaObj, "request", "body")
+	jtdToJsonSchema(schemaObj, "request", "qs")
+	jtdToJsonSchema(schemaObj, "response", "body")
+
+	return schemaObj
+}
+
+func jtdToJsonSchema(schemaObj handlerSchema, reqRes string, loc string) {
+	schema := schemaObj[reqRes][loc]
+	if schema != nil {
+		schema["type"] = "object"
+		schema["additionalProperties"] = false
+		keys := getKeys(schema["properties"])
+		if keys != nil {
+			schema["required"] = keys
+		}
+	}
+}
```

Comment:
that's probably a stupid question, but what does `jtdToJsonSchema` do? i.e. it doesn't return anything, is `schemaObj` modified by reference somehow (which is also not clear for me)?

Reply:
jtdToJsonSchema converts from JTD/JDDF (new spec, JSON Type Definition) to JSON Schema  that is more verbose and there is a library for it. I couldn't make go library from JDDF work.

Reply:
yes, maps and slices are passed by reference in go.